### PR TITLE
SASS-10306 

### PR DESCRIPTION
**Pull request by garytate123**
### Description

Please include a brief summary of the change you've made and the issue that it fixes.

Add a link to the relevant story in Jira
[SASS-CHANGE-ME](ADD URL HERE)

##### Before Reviewing

- [ ]  Have you pulled the branch down?
- [ ]  Have you assigned yourself to the PR?
- [ ]  Have you moved the task to “in review” on JIRA?
- [ ]  Have you checked to ensure all dependencies are up to date?
- [ ]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing

- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [ ]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing

- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request?
  i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [ ]  Have you run the smoke tests? (where applicable)
- [ ]  Have you run the performance tests? (where applicable)
- [ ]  Have you addressed warnings where appropriate?
- [ ]  Have you rebased against the current version of main?
- [ ]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised

- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

File: it/test/connectors/SectionCompletedConnectorISpec.scala [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+          withStatus(status).
+          withBody(body)
+      )
+    )
+
+  implicit private lazy val hc: HeaderCarrier = HeaderCarrier()
+  private lazy val connector: SectionCompletedConnector = app.injector.instanceOf[SectionCompletedConnector]
+  lazy val httpClient: HttpClientV2 = app.injector.instanceOf[HttpClientV2]
+
+  //  def appConfig(host: String): AppConfig = new AppConfig(app.injector.instanceOf[ServicesConfig]) {
```
Comment (baha-aboukoura) [inline it/test/connectors/SectionCompletedConnectorISpec.scala:52]: commented out code

**Review by baha-aboukoura — CHANGES_REQUESTED** [review]

**Review by stephenheath101 — APPROVED** [review]

**Review by baha-aboukoura — APPROVED** [review]

File: app/controllers/SectionCompletedStateController.scala [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      )
+  }
+
+  private def saveAndRedirect(answer: Boolean, taxYear: Int, benefitType: BenefitType, mtditid: String)(implicit hc: HeaderCarrier): Future[Result] = {
+    val status: JourneyStatus = if (answer) Completed else InProgress
+    val model = JourneyAnswers(mtditid, taxYear, benefitType.toString, Json.obj({
+      "status" -> status
+    }), Instant.now)
+    sectionCompletedService.set(model)
+    Future.successful(Redirect(appConfig.incomeTaxSubmissionOverviewUrl(taxYear)))
```
Comment (basotia) [inline app/controllers/SectionCompletedStateController.scala:85]: should this not be `Future.successful(Redirect(appConfig.commonTaskListUrl(taxYear)))`

File: app/controllers/SectionCompletedStateController.scala [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        formWithErrors => Future.successful(BadRequest(view(formWithErrors, taxYear, benefitType))),
+        answer => {
+         saveAndRedirect(answer, taxYear, benefitType, user.user.mtditid)
+        }
+      )
+  }
+
+  private def saveAndRedirect(answer: Boolean, taxYear: Int, benefitType: BenefitType, mtditid: String)(implicit hc: HeaderCarrier): Future[Result] = {
+    val status: JourneyStatus = if (answer) Completed else InProgress
+    val model = JourneyAnswers(mtditid, taxYear, benefitType.toString, Json.obj({
```
Comment (matthewwardcap) [inline app/controllers/SectionCompletedStateController.scala:81]: Should be benefitType.typeName currently

However we might add new attributes to the BenefitType model so just hold off at the moment

Comment (matthewwardcap) [reply app/controllers/SectionCompletedStateController.scala:81]: We might need to hear back off team-b first but will discuss monday

**Review by stephenheath101 — DISMISSED** [review]

**Review by basotia — DISMISSED** [review]

**Review by matthewwardcap — APPROVED** [review]

**Pull request merged**